core/domain/collection_services.py

```diff
@@ -936,39 +936,51 @@ def update_collection(
             committer_id, utils.get_current_time_in_millisecs())
 
 
-def create_collection_summary(collection_id, contributor_id_to_add):
+def create_collection_summary(
```

Comment:
Instead of doing all this stuff, why not just regenerate the whole thing?

(Collection summaries are transient models, subject to regeneration from scratch. Ditto for exploration summaries, topic summaries, etc.)

Reply:
Done. We skip the users that are deleted elsewhere.